linkcheck/parser/__init__.py

```diff
@@ -131,4 +131,4 @@ def parse_itms_services(url_data):
             break
 
 
-from .sitemap import parse_sitemap, parse_sitemapindex
+from .sitemap import parse_sitemap, parse_sitemapindex  # noqa: E402,F401
```

Comment:
I'm seeing a lot of E402.  I've no idea what it means, but maybe let's put it in the `extends-ignore =` instead of sprinkling noqa comments everywhere?

I remember what F401 means (unused import).  Sometimes you can avoid it by including the imported name in `__all__ = [...]`, to signify explicit re-exports.

Reply:
agreed, i had the same thought

Reply:
E402 module level import not at top of file

I got rid of the majority of the `# noqa`, listing per-file-ignores for these. I left them when, like the above, they are creating aliases for other members of the subpackage to show those lines really are there for a reason.

I have found a fair bit of dead code with these (not always imports alone), indeed I just introduced one in the recent PR for the http zero length html fix, so I would be loathe to disable them globally.